### ISSUE TITLE
the email pusher is broken in 0.33.7

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#4065]
If you leave `email_templates` unset, or set to `res/templates`, you get the following error when the pusher tries to send an email:

```
Traceback (most recent call last):
  File synapse/push/emailpusher.py, line 125, in _process
  File /home/matrix/.synapse/local/lib/python2.7/site-packages/twisted/internet/defer.py, line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File /home/matrix/.synapse/local/lib/python2.7/site-packages/twisted/python/failure.py, line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File synapse/push/emailpusher.py, line 177, in _unsafe_process
  File /home/matrix/.synapse/local/lib/python2.7/site-packages/twisted/internet/defer.py, line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File /home/matrix/.synapse/local/lib/python2.7/site-packages/twisted/python/failure.py, line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File synapse/push/emailpusher.py, line 287, in send_notification
  File /home/matrix/.synapse/local/lib/python2.7/site-packages/twisted/internet/defer.py, line 1418, in _inlineCallbacks
    result = g.send(result)
  File synapse/push/mailer.py, line 178, in send_notification_mail
  File /home/matrix/.synapse/local/lib/python2.7/site-packages/jinja2/environment.py, line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File /home/matrix/.synapse/local/lib/python2.7/site-packages/jinja2/environment.py, line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File synapse/res/templates/notif_mail.html, line 5, in top-level template code
  File /home/matrix/.synapse/local/lib/python2.7/site-packages/jinja2/loaders.py, line 235, in get_source
    raise TemplateNotFound(template)
TemplateNotFound: mail.css
```

**Comment by richvdh** [comment]
A workaround for now is to explicitly set `email_templates` to the full path of wherever `synapse/res/templates` is on your system.

**Comment by richvdh** [comment]
This was introduced by https://github.com/matrix-org/synapse/pull/4052.

So, some backstory is necessary here:

The default config (before #4052) gave an example setting of `res/templates` for `email_templates`. However, *this never worked*, because synapse changes directory to `/` once it starts (at least if it is run with the `--daemonize` arg, as synctl does). In practice therefore, anyone who uses email notifications will have set `email_templates` to an absolute path.

In #4052, I had the great idea of solving #4014 and https://github.com/matrix-org/package-synapse-debian/issues/37 by including the example templates as package data, and using them from there if `email_templates` was unset (or, in an attempt to preserve backwards-compatibility, if it was set to `res/templates` - not that that will have helped anyone in practice, as per the above).

However, it turns out that attempting to load the email templates from the package data has exactly the same problem as loading them from a relative path: the supporting templates like `mail.css` can't be found once synapse changes directory.

**Comment by richvdh** [comment]
fixed by #4069
